File: Azure Management Certificate/azure_mgmt_cert.py
# Using Azure Management Libraries with a Certificate:
from azure.common.credentials import ServicePrincipalCredentials
from azure.mgmt.resource import ResourceManagementClient

tenant_id = "your_tenant_id"
client_id = "your_client_id"
certificate_path = "/path/to/your/certificate.pem"

credentials = ServicePrincipalCredentials(
    client_id=client_id,
    secret=None,
    tenant=tenant_id,
    certificate_path=certificate_path,
)

subscription_id = "your_subscription_id"
resource_client = ResourceManagementClient(credentials, subscription_id)

# List all resource groups
for rg in resource_client.resource_groups.list():
    print(rg.name)




# Using Azure Key Vault Client:
from azure.keyvault import KeyVaultClient
from azure.common.credentials import ServicePrincipalCredentials

tenant_id = "your_tenant_id"
client_id = "your_client_id"
certificate_path = "/path/to/your/certificate.pem"

credentials = ServicePrincipalCredentials(
    client_id=client_id,
    secret=None,
    tenant=tenant_id,
    certificate_path=certificate_path,
)

key_vault_client = KeyVaultClient(credentials)
vault_url = "https://your-vault-name.vault.azure.net/"

# Retrieve a secret
secret = key_vault_client.get_secret(vault_url, "your-secret-name", "")
print(secret.value)





# Using Azure Storage Management Client:
from azure.common.credentials import ServicePrincipalCredentials
from azure.mgmt.storage import StorageManagementClient

tenant_id = 'your_tenant_id'
client_id = 'your_client_id'
certificate_path = '/path/to/your/certificate.pem'

credentials = ServicePrincipalCredentials(
    client_id=client_id,
    secret=None,
    tenant=tenant_id,
    certificate_path=certificate_path
)

subscription_id = 'your_subscription_id'
storage_client = StorageManagementClient(credentials, subscription_id)

# List all storage accounts
for account in storage_client.storage_accounts.list():
    print(account.name)

File: Azure Management Certificate/azure_mgmt_cert.js
// Using Azure Management Libraries with a Certificate:
const msRestNodeAuth = require("@azure/ms-rest-nodeauth");
const { ResourceManagementClient } = require("@azure/arm-resources");

const clientId = "your_client_id";
const tenantId = "your_tenant_id";
const subscriptionId = "your_subscription_id";
const certificatePath = "/path/to/your/certificate.pem";

msRestNodeAuth.loginWithServicePrincipalSecretWithCertificate(
    clientId,
    certificatePath,
    tenantId,
    (err, credentials) => {
        if (err) {
            console.error(err);
            return;
        }

        const client = new ResourceManagementClient(credentials, subscriptionId);
        
        client.resourceGroups.list().then((result) => {
            console.log(result);
        }).catch((err) => {
            console.error(err);
        });
    }
);



// Using Azure Key Vault Client:
const msRestNodeAuth = require("@azure/ms-rest-nodeauth");
const { KeyVaultClient } = require("@azure/keyvault");

const clientId = "your_client_id";
const tenantId = "your_tenant_id";
const certificatePath = "/path/to/your/certificate.pem";
const vaultUrl = "https://your-vault-name.vault.azure.net/";

msRestNodeAuth.loginWithServicePrincipalSecretWithCertificate(
    clientId,
    certificatePath,
    tenantId,
    (err, credentials) => {
        if (err) {
            console.error(err);
            return;
        }

        const client = new KeyVaultClient(credentials);
        
        client.getSecret(vaultUrl, 'your-secret-name', '').then((result) => {
            console.log(result.value);
        }).catch((err) => {
            console.error(err);
        });
    }
);



// Using Azure Storage Management Client:
const msRestNodeAuth = require("@azure/ms-rest-nodeauth");
const { StorageManagementClient } = require("@azure/arm-storage");

const clientId = "your_client_id";
const tenantId = "your_tenant_id";
const subscriptionId = "your_subscription_id";
const certificatePath = "/path/to/your/certificate.pem";

msRestNodeAuth.loginWithServicePrincipalSecretWithCertificate(
    clientId,
    certificatePath,
    tenantId,
    (err, credentials) => {
        if (err) {
            console.error(err);
            return;
        }

        const client = new StorageManagementClient(credentials, subscriptionId);
        
        client.storageAccounts.list().then((result) => {
            console.log(result);
        }).catch((err) => {
            console.error(err);
        });
    }
);